CMSys.UI/Controllers/UserController.cs

```diff
@@ -73,9 +75,71 @@ public IActionResult IndexUsers(int page, int perPage, string fullName)
             foreach (var item in mappedUsers.Items)
             {
                 users.Add(item);
+                var filePath = $"../CMSys.UI/wwwroot/img/{item.FullName}.png";
+                using (var ms = new MemoryStream(item.Photo))
+                    FileWriter.WriteBytesToFile(filePath, item.Photo);
             }
 
             return View(mappedUsers);
         }
+        [Authorize]
+        [HttpGet]
+        [Route("admin/users/{id}")]
+        public IActionResult GetUser(Guid id)
+        {
+            var userViewModel = new UserViewModel();
+            var user = _context.UserRepository.Find(x => x.Id == id);
+            // var userRoles = _context.RoleRepository.Filter(x => x.)
+
+            var mappedUser = _mapper.Map(user, userViewModel);
+            return View(mappedUser);
+        }
+        [Authorize]
+        [Route("admin/users/update/{id}")]
+        public IActionResult UpdateUserForm(UserViewModel userViewModel)
+        {
+            var user = _context.UserRepository.Find(x => x.Id == userViewModel.Id);
+            userViewModel = _mapper.Map(user, userViewModel);
+
+
+
+            var roles = _context.RoleRepository.All().ToList();
+
+            foreach (var role in roles)
+            {
+                if (!userViewModel.Roles.Any(x => x.Id == role.Id))
+                {
+                    userViewModel.RolesSelection.Add(new SelectListItem(role.Name, role.Id.ToString()));
+                }
+            }
+
+            return View(userViewModel);
+        }
+        [Authorize]
+        [HttpPost]
+        [Route("admin/users/update/{id}")]
+        public IActionResult UpdateUser(UserViewModel userViewModel)
```

Comment:
I do not understand what this method does and how it works. Please rewrite it based on the comments 
below.